src/compas_fab/ghpython/reachability_artist.py

```diff
@@ -0,0 +1,85 @@
+from compas.geometry import Frame
+from compas.colors import ColorMap
+from compas.artists import PrimitiveArtist
+from ghpythonlib.treehelpers import list_to_tree
+
+
+class ReachabilityMapArtist(object):  # how base on GHArtist without error?
+    """Artist for drawing a reachability map.
+
+    Parameters
+    ----------
+    reachability_map : :class:`compas.robots.ReachabilityMap`
+        Robot model.
+    **kwargs : dict, optional
+        Additional keyword arguments.
+    """
+
+    def __init__(self, reachability_map, **kwargs):
+        self.reachability_map = reachability_map
+
+    def draw_frames(self):
+        """Returns the frames of the reachability map.
+        """
+
+        def convert_to_xframe(f, flist):
```

Comment:
This function is a bit confusing to read, I think it's doing two things at the same time, one is flattening the list into 1D, and the other is converting to rhino planes. It's not clear to me if the list will allow a mixed case in which some elements are not nested and some are lists nested in the main list, I assume that is not the case and it's just one way or the other depending on the input, but never mixed in one list. If that assumption is correct, splitting this into two steps (1) flatten using `itertools.chain(*flist)` and 2) do another pass to draw rhino planes) would make this more readable.